engine/conditions.go

```diff
@@ -23,48 +34,113 @@ type Rule struct {
 }
 
 type AndCondition struct {
-	Conditions []Conditional
```

Comment:
Do we want to add context gathering to the conditions inside and/or?

Reply:
I think you are saying that inside of the or/and that we will add `As` blocks.

My initial reaction is no because it will make things more complicated and make it easier to overshadow contexts. 

If you want to add to the context, you must use a chain. To be specific, I will add a test that checks for this.

Reply:
and example is:


```yaml
- message: POM XML dependencies
  when:
    chain:
    - or:
      - chain:
        - builtin.file: pom.xml
          as: poms
        - builtin.filecontent: TODO(\(.*\))?:?(.*)?
      - builtin.xml:
          xpath: "//dependencies/dependency"
          filepaths: '{{ index .poms "filepaths" }}'
        from: poms
```

Reply:
this is the golang output:

```
 when:
      chain:
      - from: ""
        as: ""
        providerspecificconfig:
          or:
          - from: ""
            as: ""
            providerspecificconfig:
              chain:
              - from: ""
                as: poms
                providerspecificconfig:
                  client: {}
                  capability: file
                  conditioninfo: pom.xml
              - from: ""
                as: ""
                providerspecificconfig:
                  client: {}
                  capability: filecontent
                  conditioninfo: TODO(\(.*\))?:?(.*)?
          - from: poms
            as: ""
            providerspecificconfig:
              client: {}
              capability: xml
              conditioninfo:
                filepaths: '{{ index .poms "filepaths" }}'
                xpath: //dependencies/dependency
```